core/src/main/scala/akka/kafka/scaladsl/Consumer.scala

```diff
@@ -161,6 +161,18 @@ object Consumer {
                               subscription: Subscription): Source[CommittableMessage[K, V], Control] =
     Source.fromGraph(ConsumerStage.committableSource[K, V](settings, subscription))
 
+  /**
+   * The `committableSourceWithMetadata` makes it possible to add additional metadata (in the form of a string)
+   * when an offset is committed based on the record. This can be useful (for example) to store information about which
+   * node made the commit, what time the commit was made, the timestamp of the record etc.
+   */
+  def committableSourceWithMetadata[K, V](
```

Comment:
Would there be a better name for this construct?
eg. `commitWithMetadataSource`?